pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -266,6 +266,7 @@ protected void createColumnV1Indices(String column)
     int dictionaryElementSize = 0;
 
     Object sortedArray;
+    final int cardinality = 1;
```

Comment:
I might have missed something, but why is this set to 1? And where do we set the cardinality correctly when it does have dictionary columns? Am I correct in assuming that we just added the cardinality field, and it wasn't there before?